### PR TITLE
Integration: Dashboard Event Types Page

### LINKED ISSUE #18: IN: Dashboard
Integrate user's meetings controller

### LINKED ISSUE #58: Bug: UserContext Provider not loading fast enough to allow components to render with data provided to them in Dashboard (Event Types)
I'm getting stuck on getting the data from UserProvider.context to display user data on the dashboard because it's giving me a similar undefined error I got on the New Events page. The difference here is that the page expects to load the data right away as opposed to having to click a button to retrieve the data as in the New Events page.

Error Message:
```js
TypeError: userData.user is undefined
EventTypes
src/pages/dashboard/EventTypes.js:53

  50 | </Grid>
  51 | <Grid container direction="column" item xs={11}>
  52 |   <Grid item>
> 53 |     <Typography>{userData.user.firstName} {userData.user.lastName}</Typography>
     | ^  54 |   </Grid>
  55 |   <Grid item>
  56 |     <Typography color="primary">calendly.com/{userData.user.firstName.toLowerCase()}-{userData.user.lastName.toLowerCase()}</Typography>
```
Is there a react method that waits for context data to load before rendering the components from using it? I tried using a ternary operation below to check if undefined:
```js 
<Typography>{typeof userData === undefined ? "Loading" : userData.user.firstName} {typeof userData === undefined ? "Loading" : userData.user.lastName}</Typography>
```
but that isn't making the error go away. I tried looking on stack overflow, but not finding anything that can help at the moment.

### PR DESCRIPTION
**Pull request by singhmi4**
### What it Does
* user name is being pulled from context object
* meeting types are pulled from a back-end route through an axios get request and `request.params.userID` in the EventTypes.js route and rendered as Cards
* NewEventTypes.js redirects back to EventTypes.js dashboard page

### What to Look For
* proper use of axios get request to back-end
* proper use of useEffect second argument array of dependencies
* need to hit refresh after being redirected back to Dashboard after submitting new event type because it doesn't show on the page automatically

### What to Avoid
* calendapp.com/user-name not actually being pulled from user-name, but from first and last name concatenated together

### CODE REVIEW

**Comment by singhmi4** [comment]
Trying to fix the issue right now of needing to refresh the dashboard in order to see the new "event type" added to the page after the redirect so I converted this PR to draft.

**Comment by singhmi4** [comment]
Just going to branch off this branch to fix the issue I mentioned above.

File: client/src/pages/dashboard/EventTypes.js [hunk 1]
```diff
@@ -45,10 +60,14 @@ const EventTypes = () => {
               </Grid>
               <Grid container direction="column" item xs={11}>
                 <Grid item>
-                  <Typography>John Doe</Typography>
+                  {userData.user &&
```
Comment (bonnieli) [inline client/src/pages/dashboard/EventTypes.js:63]: you might want to move this logic up and wrap it around the entire block for example

```
{userData.user && (
<Grid container direction="column" item xs={11}>
   <Grid item>
      <Typography>{userData.user.firstName} {userData.user.lastName}</Typography>
   </Grid>
   <Grid item>
       <Typography color="primary">calendly.com/{userData.user.firstName.toLowerCase()}-{userData.user.lastName.toLowerCase()}</Typography>
...
)}
```

Comment (bonnieli) [reply client/src/pages/dashboard/EventTypes.js:63]: so that you don't have to repeat this logic check every time :)

**Review by bonnieli — APPROVED** [review]

**Comment by bonnieli** [comment]
Closing this before it's addressed in https://github.com/hatchways/team-vampire/pull/63

**Pull request closed without merging**